cpp/stdlib.h

```diff
@@ -118,6 +120,44 @@ void execve(Str* argv0, List<Str*>* argv, Dict<Str*, Str*>* environ);
 
 void kill(int pid, int sig);
 
+List<Str*>* listdir(Str* path);
+
+// sys/stat.h defines these macros for backwards compatability reasons.  We
+// undefine them here to avoid collisions with stat_result's members. We
+// shouldn't have any code that depends on these macros, so this should be
+// safe...
+#ifdef st_atime
+  #undef st_atime
+  #undef st_mtime
+  #undef st_ctime
+#endif
+
+class stat_result {
+ public:
+  explicit stat_result(const struct stat* st);
+
+  GC_OBJ(header_);
+
+  // These three fields are seemingly unused, but are preserved for
+  // compatability with pyext.
+  int n_fields;
+  int n_sequence_fields;
+  int n_unnamed_fields;
+
+  int st_mode;
+  int st_ino;
+  int st_dev;
+  int st_nlink;
+  int st_uid;
+  int st_gid;
+  int st_size;
+  int st_atime;
+  int st_mtime;
+  int st_ctime;
```

Comment:
these _should_ be  `unsigned long`, but i'm not exactly sure how to honor that without being inconsistent with the `stat_results` type defined in `pyext/posix_.pyi`... any ideas?

Reply:
Oh sorry I didn't respond to this comment earlier.

Yeah the integer types are one case where our translation is a little weird.

I guess these can overflow pretty easily?  We probably don't want that bug

Actually it seems like Python gives a float!

We are hashing a float in the Python code, which is bad!  doh.  Let me think about this


```
>>> import os
>>> st=os.stat("README.md")
>>> st
posix.stat_result(st_mode=33204, st_ino=6301016, st_dev=66307, st_nlink=1, st_uid=1000, st_gid=1000, st_size=10657, st_atime=1673379783, st_mtime=1665704712, st_ctime=1665704712)
>>> st.st_mtime
1665704712.5696013
```